### PR TITLE
tests: More VkBufferObj cleanup

### PR DESCRIPTION
**Pull request by spencer-lunarg**
part 2 of https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/6248

### CODE REVIEW

File: tests/framework/binding.h [hunk 1]
```diff
@@ -463,8 +463,8 @@ class Buffer : public internal::NonDispHandle<VkBuffer> {
     void destroy() noexcept;
 
     // vkCreateBuffer()
-    void init(const Device &dev, const VkBufferCreateInfo &info, VkMemoryPropertyFlags mem_props, void *alloc_info_pnext = nullptr);
-    void init(const Device &dev, const VkBufferCreateInfo &info) { init(dev, info, 0); }
+    void init(const Device &dev, const VkBufferCreateInfo &info, VkMemoryPropertyFlags mem_props = 0,
+              void *alloc_info_pnext = nullptr);
     void init(const Device &dev, VkDeviceSize size, VkMemoryPropertyFlags mem_props,
```
Comment (artem-lunarg) [inline tests/framework/binding.h:468]: Would it be possible for follow up PRs to simplify even more?!

Remove this one:
```
 void init(const Device &dev, VkDeviceSize size, VkMemoryPropertyFlags mem_props, VkBufferUsageFlags usage, void *alloc_info_pnext) 
```

and to have two `init` overloads (or maybe some other variation to keep signatures more orthogonal):

```
void init(const Device &dev, const VkBufferCreateInfo &info,
         VkMemoryPropertyFlags mem_props = 0,
         void *alloc_info_pnext = nullptr);

void init(const Device &dev, VkDeviceSize size,
         VkBufferUsageFlags usage,
         VkMemoryPropertyFlags mem_props = 0,
         void *alloc_info_pnext,
         const std::vector<uint32_t> &queue_families = {})
```

Comment (spencer-lunarg) [reply tests/framework/binding.h:468]: CI failed and need to fix something, so happy to do it here in this PR

File: tests/framework/binding.h [hunk 2]
```diff
@@ -463,23 +461,12 @@ class Buffer : public internal::NonDispHandle<VkBuffer> {
...
-    void init_as_src_and_dst(const Device &dev, VkDeviceSize size, VkMemoryPropertyFlags &reqs,
-                             const std::vector<uint32_t> *queue_families = nullptr, bool memory = true) {
-        if (memory)
-            init(dev, create_info(size, VK_BUFFER_USAGE_TRANSFER_SRC_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT, queue_families), reqs);
-        else
-            init_no_mem(dev,
-                        create_info(size, VK_BUFFER_USAGE_TRANSFER_SRC_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT, queue_families));
+    void init(const Device &dev, const VkBufferCreateInfo &info, VkMemoryPropertyFlags mem_props = 0,
+              void *alloc_info_pnext = nullptr);
+    void init(const Device &dev, VkDeviceSize size, VkBufferUsageFlags usage = VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT,
```
Comment (arno-lunarg) [inline tests/framework/binding.h:466]: Personally I would love to get rid of those `init` overloads. For instance this one could be called something like `init_with_queues`, or `init_concurrent_queues`.

File: tests/framework/binding.h [hunk 3]
```diff
@@ -432,18 +432,16 @@ static constexpr NoMemT no_mem{};
...
     }
-    explicit Buffer(const Device &dev, VkDeviceSize size, VkMemoryPropertyFlags mem_props, VkBufferUsageFlags usage,
+    explicit Buffer(const Device &dev, VkDeviceSize size, VkBufferUsageFlags usage, VkMemoryPropertyFlags mem_props = 0,
                     void *alloc_info_pnext = nullptr) {
-        init(dev, size, mem_props, usage, alloc_info_pnext);
+        init(dev, size, usage, mem_props, alloc_info_pnext);
     }
     explicit Buffer(const Device &dev, const VkBufferCreateInfo &info, NoMemT) { init_no_mem(dev, info); }
-    explicit Buffer(const Device &dev, VkDeviceSize size) { init(dev, size, 0); }
+    explicit Buffer(const Device &dev, VkDeviceSize size) { init(dev, size, VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT); }
```
Comment (arno-lunarg) [inline tests/framework/binding.h:444]: I would remove this constructor, and just use a default parameter for `VkBufferUsageFlags usage` in the constructor above

File: tests/framework/binding.h (hunk 2 above)
Comment (artem-lunarg) [reply tests/framework/binding.h:466]: Constructors also have problem with naming  variety.

**Review by arno-lunarg — APPROVED** [review]
LGTM, 2 suggestions

File: tests/framework/binding.h (hunk 2 above)
Comment (spencer-lunarg) [reply tests/framework/binding.h:466]: agree, this PR is already a pain as it touches so many tests, will be a good follow-up PR to give these better names

**Review by artem-lunarg — APPROVED** [review]
LGTM. That's a very welcomed update to buffer helper.

**Pull request merged**